rpc/src/services/protocol/proto_004/rights_service.rs

```diff
@@ -93,22 +107,48 @@ pub(crate) fn get_baking_rights(
 
     let timestamp = parameters.block_timestamp();
     let block_level = parameters.block_level();
+    let cycle_position = *rights_metadata.block_cycle_position();
+
+    // build a reverse map of rols so we have access in O(1)
+    // let rolls_map: HashMap<i32, String> = HashMap::new();
+    let rolls_map: HashMap<i32, String> = cycle_meta_data
+        .rolls_data()
+        .iter()
+        .map(|(delegate, rolls)| {
+            rolls
+                .iter()
+                .map(|roll| {
+                    (
+                        *roll,
+                        SignaturePublicKeyHash::from_tagged_bytes(delegate.clone())
+                            .unwrap()
```

Comment:
@adonagy 
unwrap?

Reply:
Oh, this is not good, will have to rework this

Reply:
@adonagy 
maybe if you use `into_iter` , you dont need to `delegate.clone()`  ?